### DISCUSSION TITLE
mongodb createAt default

### DISCUSSION THREAD

**Discussion by devmnj** [prisma/prisma#13227] [Q&A]
I have a schema which in which I used MongoDB. The default date in the field **createdAt** is not working. 

### Package JSON
``` 
{
  "name": "primsma-nextjs-blog",
  "version": "0.1.0",
  "private": true,
  "scripts": {
    "dev": "next dev",
    "build": "next build",
    "start": "next start",
    "lint": "next lint"
  },
  "dependencies": {
    "@prisma/client": "^3.13.0",
    "next": "12.1.5",
    "react": "18.1.0",
    "react-dom": "18.1.0",
    "react-markdown": "^8.0.3",
    "react-query": "^3.38.1"
  },
  "devDependencies": {
    "@types/node": "^17.0.31",
    "@types/react": "^18.0.8",
    "autoprefixer": "^10.4.7",
    "eslint": "8.14.0",
    "eslint-config-next": "12.1.5",
    "postcss": "^8.4.13",
    "prisma": "^3.13.0",
    "tailwindcss": "^3.0.24",
    "ts-node": "^10.7.0",
    "typescript": "^4.6.4"
  },
  "description": "This is a [Next.js](https://nextjs.org/) project bootstrapped with [`create-next-app`](https://github.com/vercel/next.js/tree/canary/packages/create-next-app).",
  "main": "next.config.js",
  "keywords": [],
  "author": "",
  "license": "ISC"
}
``` 

### Schema.prisma 
```
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "mongodb"
  url      = env("DATABASE_URL")
}

model Author {
  id         String @id @default(auto()) @map("_id") @db.ObjectId
  name       String
  profession String
  posts Post[]
}

model Category {
  id   String @id @default(auto()) @map("_id") @db.ObjectId
  name String
  postIds String[] @db.ObjectId
  posts Post[] @relation(fields:[postIds], references:[id])
}

model Post {
  id          String @id @default(auto()) @map("_id") @db.ObjectId
  createdAt   DateTime @default(now()) 
  updatedAt   DateTime @updatedAt 
  slug        String?            
  summary     String?
  content     String
  title_image String?
  author      Author @relation(fields:[authorId],references:[id])
  authorId    String @db.ObjectId
  categoryIds String[] @db.ObjectId
  categories  Category[] @relation(fields:[categoryIds], references:[id])
  viewCount   Int? @default(0) 
}

model User {
  id String @id @default(auto()) @map("_id") @db.ObjectId
}
``` 

If any body have a good hand , help me.

**Comment by janpio** [comment]
So this is about `  createdAt   DateTime @default(now()) ` not being filled? Or being filled with wrong information?

**Reply by devmnj** [reply]
createdAt value default  not working, that is the issue

**Reply by indicozy** [reply]
It's working for many, could you provide more information about your OS and where is your MongoDB server placed (local, Atlas)?

**Reply by devmnj** [reply]
OS: Windows  11
localdb

**Reply by devmnj** [reply]
@janpio  There is no default value get filled

**Reply by janpio** [reply]
Did you figure out the problem?

**Reply by devmnj** [reply]
@janpio  not yet

**Reply by hedaukartik** [reply]
@devmnj this is what I found https://www.prisma.io/docs/reference/api-reference/prisma-schema-reference#now
`now()` is not supported in `mongodb`.

**Reply by ChayaninSuatap** [reply]
> @devmnj this is what I found https://www.prisma.io/docs/reference/api-reference/prisma-schema-reference#now `now()` is not supported in `mongodb`.

Doc said it is implemented at "Prisma level". Not sure what that means though.

For me `createdAt DateTime @default(now()) ` is working on my local machine. (Backened running on local. DB hosted on Atlas)
But when I deploy the same app at Netlify, it's not working anymore. The field "createdAt" does not appear on mongo document and no error is reported on logs.

**Reply by pantharshit00** [reply]
Hello! Sorry for late followup here. 

> Doc said it is implemented at "Prisma level". Not sure what that means though.

It means Prisma engine will populate the value of that field instead of the database doing it. 


I wasn't able to reproduce this in the latest version of Prisma. Can you please let me know if this is still relevant?

**Reply by ChayaninSuatap** [reply]
I have deployed the production app on my own server and it is working normally. So I guess the issue appeared because I did not config the Netlify setting correctly. I just don't have time to dig it deeper. Anyway I think it is working now. Thanks for checking this out.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
